pkg/kdk/init.go

```diff
@@ -0,0 +1,207 @@
+	"github.com/cisco-sso/kdk/pkg/prompt"
+	"github.com/cisco-sso/kdk/pkg/ssh"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/go-connections/nat"
+	"github.com/ghodss/yaml"
+)
+
+// TODO (rluckie) refactor to be a method of kdkConfig type
+func InitKdkConfig(cfg KdkEnvConfig, logger logrus.Entry) error {
+
+	// Initialize storage mounts/volumes
+	var mounts []mount.Mount         // hostConfig
+	volumes := map[string]struct{}{} // containerConfig
+	labels := map[string]string{"kdk": Version}
+
+	// Define mount configurations for mounting the ssh pub key into a tmp location where the bootstrap script may
+	//   copy into <userdir>/.ssh/authorized keys.  This is required because Windows mounts squash permissions to
+	//   777 which makes ssh fail a strict check on pubkey permissions.
+	source := cfg.PublicKeyPath()
+	target := "/tmp/id_rsa.pub"
+	mounts = append(mounts, mount.Mount{Type: mount.TypeBind, Source: source, Target: target, ReadOnly: true})
+	volumes[target] = struct{}{}
+
+	// Keybase mounts
+	source, target, err := keybase.GetMounts(cfg.ConfigRootDir(), logger)
+	if err != nil {
+		logger.Warn("Failed to add keybase mount:", err)
+	} else {
+		mounts = append(mounts, mount.Mount{Type: mount.TypeBind, Source: source, Target: target, ReadOnly: false})
+		volumes[target] = struct{}{}
+	}
+
+	// Define Additional volume bindings
+	for {
+		prmpt := prompt.Prompt{
+			Text:     "Would you like to mount additional docker host directories into the KDK? [y/n] ",
+			Loop:     true,
+			Validate: prompt.ValidateYorN,
+		}
+		if result, err := prmpt.Run(); err == nil && result == "y" {
+			prmpt = prompt.Prompt{
+				Text:     "Please enter the docker host source directory (e.g. /Users/<username>/Projects) ",
+				Loop:     true,
+				Validate: prompt.ValidateDirExists,
+			}
+			source, err := prmpt.Run()
+			if err == nil {
+				logger.Infof("Entered host source directory mount %v", source)
+			}
+
+			prmpt = prompt.Prompt{
+				Text:     "Please enter the docker container target directory (e.g. /home/<username>/Projects) ",
+				Loop:     false,
+				Validate: nil,
+			}
+			target, err := prmpt.Run()
+			if err == nil {
+				logger.Infof("Entered container target directory mount %v", target)
+			}
+
+			mounts = append(mounts, mount.Mount{Type: mount.TypeBind, Source: source, Target: target, ReadOnly: false})
+			volumes[target] = struct{}{}
+		} else {
+			break
+		}
+	}
+
+	// Create the Default configuration struct that will be written as the config file
+	cfg.ConfigFile.ContainerConfig = &container.Config{
+		Hostname: cfg.ConfigFile.AppConfig.Name,
+		Image:    cfg.ImageCoordinates(),
+		Tty:      true,
+		Env: []string{
+			"KDK_USERNAME=" + cfg.User(),
+			"KDK_SHELL=" + cfg.ConfigFile.AppConfig.Shell,
+			"KDK_DOTFILES_REPO=" + cfg.ConfigFile.AppConfig.DotfilesRepo,
+		},
+		ExposedPorts: nat.PortSet{
+			"2022/tcp": struct{}{},
```

Comment:
I'm not familiar with `struct{}{}`. Can you explain so I can learn?

Reply:
struct{} is a type.  struct{}{} is an instantiation of the struct{} type.

I'm no go expert, so I may be incorrect.